packages/web-components/src/components/date-picker/date-picker.scss

```
@@ -77,7 +77,7 @@ $css--plex: true !default;
     display: none;
   }
 }
-:host(#{$prefix}-date-picker-input[warn]) {
+:host(#{$prefix}-date-picker-input[warn]:not([disabled]):not([readonly])) {
```

```suggestion
:host(#{$prefix}-date-picker-input[warn]:not([invalid])) {
```

Seems the change was reverted